crates/sui-core/src/transaction_outputs.rs

```diff
@@ -202,6 +209,46 @@ impl TransactionOutputs {
         std::mem::take(&mut *self.accumulator_events.lock())
     }
 
+    /// If this transaction is an accumulator settlement transaction, return the balance settlement info.
+    /// Otherwise, return None.
+    /// Note that we need to return Some even when the balance changes are empty.
+    /// This is important since the balance withdraw scheduler expects to see every accumulator
+    /// version get settled.
+    pub fn get_balance_settlement_info(&self) -> Option<BalanceSettlement> {
+        if !self.written.contains_key(&SUI_ACCUMULATOR_ROOT_OBJECT_ID) {
+            return None;
+        }
+        let accumulator_version = self.effects.lamport_version();
+
+        let accumulator_events = self.accumulator_events.lock();
+        let mut balance_changes = BTreeMap::new();
+        for event in accumulator_events.iter() {
+            let TypeTag::Struct(struct_ty) = &event.write.address.ty else {
+                continue;
+            };
+            if !Balance::is_balance(struct_ty) {
+                continue;
+            }
+
+            let value = match event.write.value {
+                AccumulatorValue::Integer(value) => value,
+                AccumulatorValue::IntegerTuple(..) => {
+                    panic!("Unexpected accumulator value: {:?}", event.write.value)
+                }
+            };
+            let change = match event.write.operation {
```

Comment:
settlement transactions don't actually emit accumulator events... I think what we need to do here is read the accumulator objects that were written. What may be even simpler is to have a native call that settlement transactions call to notify settlements